### DISCUSSION TITLE
How to make code server jupyter support http

### DISCUSSION THREAD

**Discussion by Pythoneryeah** [coder/code-server#6375] [Q&A]
Hello everyone, remote device access code-server jupyter notebook failed, local device access is OK. Before the god said jupyter does not support https, can you modify the source code to make code-server support http?
![image](https://github.com/coder/code-server/assets/37642828/32854a91-9c5a-467b-8301-6e50f4d28bec)

**Comment by code-asher** [comment]
The only way around it in code that I can think of is to remove 
the use of service workers in VS Code entirely but I am not sure 
this is even possible.  Someone would need to research.

Browsers will refuse to run service workers in an insecure context 
by default but depending on your browser it might be possible to 
disable that security measure.  For example in Chrome I think you 
can go to chrome://flags/#unsafely-treat-insecure-origin-as-secure 
and enter in the URLs you want to whitelist although I have not 
tried this.

Alternatives are to get a certificate (even if self-signed) and 
add it to your browser or system or to proxy through localhost.

**Reply by Pythoneryeah** [reply]
Thank you very much. I will try it according to your suggestions

**Reply by Pythoneryeah** [reply]
“to proxy through localhost". Excuse me, how do I implement it?

**Reply by code-asher** [reply]
The way I do it is with SSH:

```
ssh -L 8080:localhost:8080 domain.tld
```

If you have no SSH access then you would need to set up a reverse proxy, maybe NGINX or Caddy or ngrok.
